xcube/core/store/storepool.py

```diff
@@ -0,0 +1,273 @@
+
+    If *store_id* is prefixed by a "@", it is an "instance identifier".
+    In this case the store instance is retrieved from the expected *store_pool* argument.
+    Otherwise a new store instance is created using optional *store_params*.
+
+    :param store_id: Store identifier, may be prefixed by a "@" to indicate a store instance identifier.
+    :param store_params: Store parameters, only valid if *store_id* is not an instance identifier.
+    :param store_pool: A pool of configured store instances used if *store_id* is an instance identifier.
+    :return: a data store instance
+    :raise: DataStoreError if a configured store does not exist
+    """
+    if store_id.startswith('@'):
+        store_instance_id = store_id[1:]
+        if store_pool is None:
+            raise ValueError(f'store_pool must be given,'
+                             f' with store_id ("{store_id}") referring to a configured store')
+        if store_params:
+            raise ValueError(f'store_params cannot be given,'
+                             f' with store_id ("{store_id}") referring to a configured store')
+        return store_pool.get_store(store_instance_id)
+    return new_data_store(store_id, **(store_params or {}))
+
+
+DATA_STORE_CONFIG_SCHEMA = JsonObjectSchema(
+    properties=dict(
+        store_id=JsonStringSchema(min_length=1),
+        store_params=JsonObjectSchema(
+            additional_properties=True
+        ),
+        name=JsonStringSchema(min_length=1),
+        description=JsonStringSchema(min_length=1),
+    ),
+    required=['store_id'],
+    # Uncomment, once needed.
+    # factory=_store_config_factory,
+    # serializer=_store_config_serializer,
+)
+
+DATA_STORE_POOL_SCHEMA = JsonObjectSchema(
+    additional_properties=DATA_STORE_CONFIG_SCHEMA,
+    # Uncomment, once needed.
+    # factory=_store_pool_factory,
+    # serializer=_store_pool_serializer,
+)
+
+
+# def _store_config_factory(**kwargs):
+#     return DataStoreConfig.from_dict(kwargs)
+#
+#
+# def _store_config_serializer(store_config: 'DataStoreConfig'):
+#     return store_config.to_dict()
+#
+#
+# def _store_pool_factory(**kwargs):
+#     return DataStorePool.from_dict(kwargs)
+#
+#
+# def _store_pool_serializer(store_pool: 'DataStorePool'):
+#     return store_pool.to_dict()
+
+class DataStoreConfig:
+    """
+    The configuration of a data store.
+    The class is used by :class:DataStorePool to instantiate stores in a deferred manner.
+
+    :param store_id: the data store identifier
+    :param store_params: optional store parameters
+    :param name: a human-readable name for the store instance
+    :param description: a human-readable description of the store instance
+    """
+
+    def __init__(self,
+                 store_id: str,
+                 store_params: Dict[str, Any] = None,
+                 name: str = None,
+                 description: str = None):
+        assert_given(store_id, name='store_id')
```

Comment:
> Why not also store_name and store_description?

Because name (or title) and description are properties of the store's _configuration_, rather than the store itself. 

`store_id` is named that way, because it is the identifier of the store (type), it is _not_ the identifier of the configuration.

> A better name than "name" would be "title"

Agree to use "title".